src/ripple/protocol/impl/STAmount.cpp

```diff
@@ -1254,7 +1264,15 @@ divRound (STAmount const& num, STAmount const& den,
     int offset = numOffset - denOffset - 17;
     canonicalizeRound (
         isXRP (issue), amount, offset, resultNegative != roundUp);
-    return STAmount (issue, amount, offset, resultNegative);
+    STAmount result (issue, amount, offset, resultNegative);
+    if (switchovers.enableUnderflowFix () && roundUp && !resultNegative && !result)
+    {
+        // return the smallest value above zero
+        amount = STAmount::cMinValue;
+        offset = STAmount::cMinOffset;
+        return STAmount (issue, amount, offset, resultNegative);
+    }
+    return result;
```

Comment:
I'm getting ultra-picky here, so this is just for your consideration.  By visual inspection these 9 new lines are identical to the nine new lines in `mulRound`.  Might it make sense to make small local inline function that captures the behavior here?  Nine duplicated lines may be below the threshold of caring, but I'll at least offer the thought.